webview/src/setup/components/dvc/Dvc.tsx

```diff
@@ -17,54 +14,49 @@ import {
 } from '../messages'
 import { EmptyState } from '../../../shared/components/emptyState/EmptyState'
 import { usePrevious } from '../../hooks/usePrevious'
+import { updateSectionCollapsed } from '../../state/webviewSlice'
+import { SetupState } from '../../store'
 
-type DvcProps = {
-  canGitInitialize: boolean | undefined
-  cliCompatible: boolean | undefined
-  dvcCliDetails: DvcCliDetails | undefined
-  isPythonExtensionUsed: boolean
-  needsGitInitialized: boolean | undefined
-  projectInitialized: boolean
-  pythonBinPath: string | undefined
-  setSectionCollapsed: (sectionCollapsed: SectionCollapsed) => void
-  hasReceivedMessageFromVsCode: boolean
-}
-
-export const Dvc: React.FC<DvcProps> = ({
-  canGitInitialize,
-  cliCompatible,
-  dvcCliDetails,
-  isPythonExtensionUsed,
-  needsGitInitialized,
-  projectInitialized,
-  pythonBinPath,
-  hasReceivedMessageFromVsCode,
-  setSectionCollapsed
-}) => {
+export const Dvc: React.FC = () => {
+  const dispatch = useDispatch()
+  const hasWebviewData = useSelector(
+    (state: SetupState) => state.webview.hasData
+  )
+  const {
+    canGitInitialize,
+    cliCompatible,
+    dvcCliDetails,
+    isPythonExtensionUsed,
+    needsGitInitialized,
+    projectInitialized,
+    pythonBinPath
+  } = useSelector((state: SetupState) => state.dvc)
```

Comment:
For now, I moved the state into the "main" section components but we can lower the amount of props being passed around further. For example `isPythonExtensionUsed` is only used by `DvcEnvCommandRow`. 

I'll take care of this in a second pr though since this one is already up to +400 lines added though.